dvc/fs/ssh.py

```diff
@@ -0,0 +1,133 @@
+import getpass
+import os.path
+import threading
+
+from funcy import cached_property, first, memoize, silent, wrap_prop, wrap_with
+
+import dvc.prompt as prompt
+from dvc.scheme import Schemes
+
+from .fsspec_wrapper import CallbackMixin, FSSpecWrapper
+
+_SSH_TIMEOUT = 60 * 30
+_SSH_CONFIG_FILE = os.path.expanduser(os.path.join("~", ".ssh", "config"))
+
+
+@wrap_with(threading.Lock())
+@memoize
+def ask_password(host, user, port):
+    return prompt.password(
+        "Enter a private key passphrase or a password for "
+        "host '{host}' port '{port}' user '{user}'".format(
+            host=host, port=port, user=user
+        )
+    )
+
+
+# pylint:disable=abstract-method
+class SSHFileSystem(CallbackMixin, FSSpecWrapper):
+    scheme = Schemes.SSH
+    REQUIRES = {"sshfs": "sshfs"}
+
+    DEFAULT_PORT = 22
+    PARAM_CHECKSUM = "md5"
+
+    def _with_bucket(self, path):
+        if isinstance(path, self.PATH_CLS):
+            path_str = path.path
+        else:
+            path_str = super()._with_bucket(path)
+        return os.path.normpath(path_str)
```

Comment:
It seems like windows failures are related with using forward-slashes, so normalizing them here resolves all the CI failures.